janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestClientSetup.java

```diff
@@ -122,7 +123,26 @@ protected RestElasticSearchClient getElasticSearchClient(RestClient rc, int scro
      * @return callback or null if the request customization is not needed
      */
     protected RequestConfigCallback getRequestConfigCallback(Configuration config) {
-        return null;
+
+        final List<RequestConfigCallback> callbackList = new LinkedList<>();
+
+        final Integer connectTimeout = config.get(ElasticSearchIndex.CONNECT_TIMEOUT);
+        final Integer socketTimeout = config.get(ElasticSearchIndex.SOCKET_TIMEOUT);
+
+        callbackList.add((requestConfigBuilder) ->
+            requestConfigBuilder.setConnectTimeout(connectTimeout).setSocketTimeout(socketTimeout));
+
+        if (callbackList.isEmpty()) {
```

Comment:
Seems this condition always evaluates to false.

Reply:
Yes, good catch, bad copy-paste...